### PR TITLE
 Bolg_App_Rails

### PR DESCRIPTION
**Pull request by AhmadiGu**
### Process data model branch
*The following changes introduced:*

- create the user model according to the UML
- create the post model according to the UML
- create the comment model according to the UML
- create the like model according to the UML

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (GzimAsani) [inline app/models/user.rb]: - [ ] According to the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png), the foreing key should be `'author_id'` example:
`has_many :posts, foreign_key: 'author_id'`

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
```
Comment (GzimAsani) [inline app/models/post.rb]: - [ ] According to the diagram this should be `belongs_to :author, class_name: 'User'`

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
```
Comment (GzimAsani) [inline app/models/comment.rb]: - [ ] According to the diagram this should be `belongs_to :author, class_name: 'User'`

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
```
Comment (GzimAsani) [inline app/models/like.rb]: - [ ] According to the diagram this should be `belongs_to :author, class_name: 'User'`

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  has_many :likes
+  has_many :comments
+
+  def posts_counter_updater(author_id)
+    user = User.find_by(id: author_id)
+    user.posts_counter = Post.where(author_id:).count
+    user.save
+  end
```
Comment (GzimAsani) [inline app/models/post.rb]: - [ OPTIONAL ] This is good implementation, but I think a more direct approach will be better to read and understand: _**Same goes for comments and likes **_
```ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @AhmadiGu                    :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your models, :+1: 

 ## Required Changes 

- [ ] Check the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by gemmen29 — APPROVED** [review]
Hi @AhmadiGu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**